src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -583,17 +583,22 @@ public void onFailure(Throwable caught) {
 		}
 	}
 
-	private Optional<Long> getVersion() {
+	/**
+	 * Retrieves the version of the entity, if it's not the latest version (i.e. the entity is immutable).
+	 *
+	 * For versionable entities, the result depends on if the entity is the latest version.
+	 * If the entity is not the latest version, the optional will contain the version number.
+	 * If the entity is the latest version, the optional will be empty.
+	 *
+	 * If the entity is not versionable, it returns an empty optional.
+	 */
+	private Optional<Long> getVersionIfNotLatest() {
 		Long version = null;
 		Entity entity = entityBundle.getEntity();
-		if (entity instanceof Versionable) {
+		if (entity instanceof VersionableEntity) {
 			VersionableEntity versionableEntity = ((VersionableEntity) entity);
-			if (versionableEntity instanceof Table
-					&& versionableEntity.getVersionLabel() != null
-					&& versionableEntity.getVersionLabel().equals("in progress")) {
-				// TODO: This is undefined behavior. Use `isLatestVersion` when PLFM-6583 is complete.

```

Comment:
The above condition didn't hold for Views (they don't have the label, `"in progress"`), which caused the issue.

Reply:
Looking for that hard-coded version label text should have been a red flag during code review, sorry I missed this.

Reply:
I explicitly told you it was there, so I think you saw it. I think I said I was sure it wouldn't break anything, and was wrong! My fault.